### PR TITLE
Autoscroll while dragging sortable-items

### PR DESCRIPTION
**Pull request by tim-evans**
This builds in functionality to scroll the container when a `{{sortable-item}}` reaches the bounding edge of its parent scroll container.

Some more work is needed to get scrolling smooth when the browser window is the element that should scroll, but it works wonderfully in scrollable containers. To test, you can try dragging items to the edges of the browser window and the page should move :)

### CODE REVIEW

File: addon/mixins/sortable-item.js [hunk 1]
```diff
@@ -328,6 +330,81 @@ export default Mixin.create({
...
+      }
+
+      if (delta !== 0) {
+        let speed = this.get('maxScrollSpeed');
+        delta = Math.min(Math.max(delta, -1 * speed), speed);
+
+        scrollContainer[scrollKey](scroll + delta);
+        if (this._event) {
+          run(() => drag(this._event));
+        }
```
Comment (jgwhite) [inline addon/mixins/sortable-item.js]: I don’t fully understand this step. Why is it necessary?

File: addon/mixins/sortable-item.js [hunk 2]
```diff
@@ -328,6 +330,81 @@ export default Mixin.create({
...
+      let scroll = scrollContainer[scrollKey]();
+
+      let delta = 0;
+      if (trailingEdge >= scrollContainer[trailingEdgeKey]) {
+        delta = trailingEdge - scrollContainer[trailingEdgeKey];
+      } else if (leadingEdge <= scrollContainer[leadingEdgeKey]) {
+        delta = leadingEdge - scrollContainer[leadingEdgeKey];
+      }
+
+      if (delta !== 0) {
```
Comment (jgwhite) [inline addon/mixins/sortable-item.js]: When the dragged item is at the edge of the window or scroll container wouldn’t the delta be exactly 0? Is that why scroll halts when you’d expect it to continue all the way to the edge?

Comment (tim-evans) [inline addon/mixins/sortable-item.js]: It would grow the container in this case. Which isn't really great

Comment (tim-evans) [inline addon/mixins/sortable-item.js]: Try commenting it out and see what happens :)

Comment (tim-evans) [inline addon/mixins/sortable-item.js]: Sorry, reading the wrong thing! This prevents too many calls to expensive functions

File: addon/mixins/sortable-item.js (hunk 1 above)
Comment (tim-evans) [inline addon/mixins/sortable-item.js]: This is so you don't need to wiggle the mouse to get the dragged item to follow along. This is the part that is best illustrated by commenting out the call to `drag` to see the difference

Comment (jgwhite) [inline addon/mixins/sortable-item.js]: Gotcha 👌

File: addon/mixins/sortable-item.js (hunk 2 above)
Comment (jgwhite) [inline addon/mixins/sortable-item.js]: Okay, I’m going to play with the code a little more to get up to speed.

**Comment by tim-evans** [comment]
Poking on this!

**Comment by jgwhite** [comment]
Hey, sorry… I'm away on wedding/honeymoon for a little while. Will try to find time soon.

**Comment by tim-evans** [comment]
Oh, you enjoy that- don't look at github!  😄 

Congrats!!

**Comment by tim-evans** [comment]
A friendly poke on this! I hope you had a fantastic 🍯  🌕

**Comment by jgwhite** [comment]
Thanks for the reminder! :D

**Comment by jgwhite** [comment]
@tim-evans so sorry! You’re not being a bother, I’m just being very lazy ;) Taking another look now.

**Comment by jgwhite** [comment]
Just noticed the project’s test suite has a pretty major incompatibility with the next version of Ember. I’m afraid review will take a little longer.

**Comment by tim-evans** [comment]
No problem :)

**Comment by acburdine** [comment]
@tim-evans @jgwhite I'm gonna work on implementing the test-suite fixes hopefully today or tomorrow - and then I will get this reviewed and merged 😄

**Comment by tim-evans** [comment]
Yay!

**Comment by acburdine** [comment]
@tim-evans can you rebase this against master? I fixed the changes for the test-suite so this should be good :)

**Comment by tim-evans** [comment]
Can do!

**Comment by tim-evans** [comment]
@acburdine rebased 👍

**Comment by acburdine** [comment]
Tested it and this looks good 👍

**Comment by jgwhite** [comment]
💥

**Pull request merged**